pkg/apis/triggers/v1alpha1/trigger_types.go

```diff
@@ -135,6 +138,74 @@ func (ti *TriggerInterceptor) defaultInterceptorKind() {
 	}
 }
 
+func (ti *TriggerInterceptor) updateCoreInterceptors() error {
+	if ti.Ref.Name != "" {
+		return nil
+	}
+	ti.Ref.Name = ti.GetName()
+	ti.Params = []InterceptorParams{}
+	switch ti.Ref.Name {
+	case "bitbucket":
+		if err := addToParams(&ti.Params, "secretRef", ti.DeprecatedBitbucket.SecretRef); err != nil {
```

Comment:
These could nil panic if ti or DeprecatedBitbucket is nil. We should add checks to prevent this (as well as for the other fields)

Reply:
Added a check for when ti is nil. 
A bit convoluted but the checks for the other fields being not nil happens in the call ti.GetName() which ensures that if the name is bitbucket, then ti.DeprecatedBitbucket is not nil: https://github.com/tektoncd/triggers/blob/main/pkg/apis/triggers/v1alpha1/trigger_types.go#L145